src/charm.py

```diff
@@ -943,7 +947,7 @@ def _open_ports_tcp(self, ports: int) -> None:
                 logger.exception("failed opening port: %s", str(e))
                 raise
 
-    def _install_snap_packages(self, packages: List[str]) -> None:
+    def install_snap_packages(self, packages: List[str]) -> None:
```

Comment:
**todo:** The type-hint is wrong here I believe.

It should be `install_snap_packages(self, packages: list[tuple[str, str, str]]): -> None`

Reply:
Or, probably better, you add a type for it.

```python
# config.py
from typing import TypeAlias

Package: TypeAlias = tuple[str, str, str]


# charm.py
from config import Package

def install_snap_packages(self, packages: Package): -> None
...

Reply:
Ah yes, let me do this. Thank you Marc 🙇